src/main/webapp/app/tasks/task-list/task-list.component.spec.ts

```diff
@@ -53,4 +54,31 @@ describe('TaskListComponent', () => {
     // then
     expect(deleteEmitter).toHaveBeenCalledWith({id: 'id', name: 'My task'});
   });
+
+  it('should filter the task using a search term', () => {
+    // given
+    component.tasks = [];
+    component.tasks.push({id: 'id1', name: 'My task 1'});
+    component.tasks.push({id: 'id2', name: 'My task 2'});
+
+    // when
+    component.searchTerms = 'task 2'
+
+    // then
+    expect(component.tasks).lengthOf(1);
+  });
+
+  it('should filter the task using a search pattern', () => {
+    // given
+    component.tasks = [];
```

Comment:
Why are you not assigning the array with the values instead of pushing them into the array afterwards?